src/test/java/org/apache/commons/imaging/roundtrip/ImageAsserts.java

```diff
@@ -99,4 +99,18 @@ static void assertEquals(final File a, final File b) throws IOException {
             Assertions.assertEquals(aByte, bByte);
         }
     }
+
+    static void assertEquals(byte[] aData, byte[] bData) {
+        for (int i = 0; i < aData.length; i++) {
+            final int aByte = 0xff & aData[i];
+            final int bByte = 0xff & bData[i];
+
+            if (aByte != bByte) {
+                Debug.debug("i: " + i);
+                Debug.debug("aByte: " + aByte + " (0x" + Integer.toHexString(aByte) + ')');
+                Debug.debug("bByte: " + bByte + " (0x" + Integer.toHexString(bByte) + ')');
+            }
+            Assertions.assertEquals(aByte, bByte);
+        }
+    }
```

Comment:
Is there a reason for using this over using JUnit's [`assertArrayEquals`](https://junit.org/junit5/docs/5.0.0/api/org/junit/jupiter/api/Assertions.html#assertArrayEquals-byte:A-byte:A-) ? The `Debug` is not active by default, I believe, so using this doesn't seem to add must value.

Reply:
I think the JUnit version should work too. Not sure if what `ImageAsserts#assertEquals(File, File)` is doing is necessary. It's reading the bytes from the stream, I assume, without handling `int` values (and thus having to apply a filter mask `0xff` to clear the other bits in the 32 byte-Java).

The point here is that if the JUnit's `assertArrayEquals` work, it's better it's less code for us to test :+1: